test/accessor_tests.cc

```diff
@@ -34,7 +34,7 @@ namespace detail {
 	}
 
 	TEST_CASE_METHOD(test_utils::runtime_fixture, "accessors mode and target deduced correctly from SYCL 2020 tag types and no_init property", "[accessor]") {
```

Comment:
This reminds me, I actually have a patch lying around that I wanted to upstream for some time; I think this test can be done purely at compile time: https://gist.github.com/psalz/bfa219820ba6f0e0ee6713d6d0e31a3c. Maybe you want to incorporate this?

Reply:
I've included the patch and also added tests for the 0-dimensional case. The `property_list` constructors now produce `write` instead of `discard_write` modes, but this should be irrelevant as they always throw anyway.